Assets/HoloToolkit/UX/Scripts/Tooltip/ToolTip.cs

```diff
@@ -0,0 +1,529 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using System;
+using UnityEngine;
+using HoloToolkit.Unity.UX;
+using HoloToolkit.Unity.Buttons;
+
+namespace HoloToolkit.UX.ToolTips
+{
+    [RequireComponent(typeof(ToolTipConnector))]
+
+    /// <summary>
+    /// Class for Tooltip object
+    /// Creates a floating tooltip that is attached to an object and moves to stay in view as object rotates with respect to the view.
+    /// </summary>
+    public class ToolTip : Button
+    {
+        [SerializeField]
+        private bool showBackground = true;
+        /// <summary>
+        /// getter/setter for showing the opaque background of tooltip.
+        /// </summary>
+        public bool ShowBackground
+        {
+            get
+            {
+                return showBackground;
+            }
+            set
+            {
+                showBackground = value;
+                GetComponent<ToolTipBackgroundMesh>().IsVisible = value;
```

Comment:
what should happen if GetComponent fails? is there a cleaner solution than a nullref exception?